### PR TITLE
App import to existing app, and public API endpoints

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
This PR solves the following issues:
- https://linear.app/budibase/issue/BUDI-7400/[existing-app-update]-endpoint-to-override-an-existing-app
- https://linear.app/budibase/issue/BUDI-7458/[existing-app-update]-frontend-for-app-update

Also has a pro component, PR: https://github.com/Budibase/budibase-pro/pull/224

This introduces the capability of importing an app to an existing app. This essentially overwrites all of the "app" components, without removing any of the user permissions/internal data from the system. All of the following will be updated:
- datasources
- queries
- table schemas
- views
- screens
- automations/webhooks

Meaning that everything that is "designed" within the builder will be replaced by what is coming from the export. It is important to note that the app export does not need to be from the same app, technically you can import from a totally different apps export, although if published this can cause some parts of the data to become inaccessible (tables/columns don't exist etc).

There isn't much else to this feature, it works about as you would expect - the frontend is quite simple, just an extension of the current "Import" section of the app settings:

image

and the modal:
image

There is also a public API component to this, which requires a business/enterprise license that can be used for automation. It is now possible to export/import apps through the public API - this also includes the OpenAPI documentation of such APIs.

### CODE REVIEW

**Review by PClmnt — APPROVED** [review]
UI LGTM

File: packages/server/src/sdk/app/backups/imports.ts [hunk 1]
```diff
@@ -151,7 +151,8 @@ export function getListOfAppsInMulti(tmpPath: string) {
 export async function importApp(
   appId: string,
   db: Database,
-  template: TemplateType
+  template: TemplateType,
+  opts: { objStore: boolean } = { objStore: true }
```
Comment (adrinr) [inline packages/server/src/sdk/app/backups/imports.ts:155]: What is this `objStore`?

File: packages/types/src/documents/document.ts [hunk 2]
```diff
@@ -39,6 +39,25 @@ export enum DocumentType {
   AUDIT_LOG = "al",
 }
 
+// these are the core documents that make up the data, design
+// and automation sections of an app. This excludes any internal
+// rows as we shouldn't import data.
+export const DocumentsToImport: DocumentType[] = [
```
Comment (adrinr) [inline packages/types/src/documents/document.ts:45]: These are the "allowed to import doc types", right?

File: packages/types/src/sdk/licensing/feature.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export enum Feature {
   SYNC_AUTOMATIONS = "syncAutomations",
   APP_BUILDERS = "appBuilders",
   OFFLINE = "offline",
-  USER_ROLE_PUBLIC_API = "userRolePublicApi",
+  EXPANDED_PUBLIC_API = "expandedPublicApi",
```
Comment (adrinr) [inline packages/types/src/sdk/licensing/feature.ts:14]: Is this just a rename? Is expanded public API the business name?

File: packages/server/src/sdk/app/applications/import.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    // get a temporary version of the import
+    // don't need obj store, the existing app already has everything we need
+    await backups.importApp(devId, tempDb, template, { objStore: false })
+    // get the documents to copy
+    const documents = await getImportableDocuments(tempDb)
+    // clear out the old documents
+    await removeImportableDocuments(appDb)
+    // now write the import documents
+    await appDb.bulkDocs(documents)
```
[the remark this answers is not in the document]
Comment (mike12345567) [reply packages/server/src/sdk/app/applications/import.ts:81]: I could actually delete and write the documents in a single call - I will do that instead as that would remove any chance of deletion then not importing.

File: packages/server/src/api/controllers/public/applications.ts [hunk 5]
```diff
@@ -91,12 +95,22 @@ export async function publish(ctx: UserCtx, next: Next) {
   })
 }
 
+// get licensed endpoints from pro
+export const importToApp = proSdk.publicApi.applications.buildImportFn(
```
[the remark this answers is not in the document]
Comment (mike12345567) [reply packages/server/src/api/controllers/public/applications.ts:99]: Responded via DMs.

File: packages/server/src/sdk/app/backups/imports.ts (hunk 1 above)
Comment (mike12345567) [reply packages/server/src/sdk/app/backups/imports.ts:155]: Its the object store, basically means whether it should try to import anything to the object store or not, theres a similar parameter on the export - I can rename them.

File: packages/types/src/documents/document.ts (hunk 2 above)
Comment (mike12345567) [reply packages/types/src/documents/document.ts:45]: Yeap, well less allowed and more desired - the ones we want, the rest we have no desire to import.

**Pull request merged**